### DISCUSSION TITLE
[BUG]'Microsoft.Extensions.DependencyInjection.ServiceCollection

### DISCUSSION THREAD

**Discussion by dinglea** [pnp/powershell#4096] [General]
if i run it in a windows all by itself there is no issues but running in a Runspace or if modules have been loaded first i am getting the error

Connect-PnPOnline -Url $SiteURL -Credentials $Cred
PowerShell: 7.4.1
Installed Version:  2.5.0

Error:  Failed to connect to https://URL. Error details: Could not load type 'Microsoft.Extensions.DependencyInjection.ServiceCollection' from assembly 'Microsoft.Extensions.DependencyInjection.Abstractions, Version=5.0.0.0, Culture=neutral

**Comment by jlrouzies-mantu** [comment]
any solution to this? get the same issue

**Comment by AndersRask** [comment]
Check out what assemblies collide and either make sure that version of those modules are older, or force the load of the conflicting module by calling the module with the affected assembly.

As an example, if your PnP.PowerShell and MgGraph have an assembly version conflict with authentication, force load the module you want to "win" the assembly conflict by connecting to graph and fetch a single user

```
try {
            # Connect Graph
            Connect-MgGraph -Identity -NoWelcome -ErrorAction Stop
            Write-Verbose -Message ("Connected to Graph!")
            # HACK: to avoid assembly load conflict with PnP, call graph first
            Get-MgUser -ConsistencyLevel eventual -Count userCount -Filter "startsWith(DisplayName, 'a')" -Top 1 | Out-Null
        }
        catch {
            throw "Error connecting to Graph"
        }
```

**Comment by jlrouzies-mantu** [comment]
Thank you, this works, indeed its conflict between Graph and Pnp

I have latest Pnp, but Graph 2.28, and I dont see a PnP documentation that says on which version it depends
